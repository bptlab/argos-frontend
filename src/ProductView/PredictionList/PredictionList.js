import React, { Component } from 'react';
import PredictionView from './PredictionView/PredictionView.js';

class PredictionList extends Component {
    
    generateErrorList() {
        const errorList = [];
        if(this.props.configuration && this.props.configuration.errorTypes) {
            this.props.configuration.errorTypes.forEach((errorType, index) => {
                errorList.push(
                    <PredictionView
                        key={index}
                        errorType={errorType} />
                );
            });
        }
        return errorList;
    }
    
    render() {
        const errorList = this.generateErrorList();
        return (
            <div className="container prediction-list-container">
                <h2>Possible product errors</h2>
                <ul className="list-group">
                    {errorList}
                </ul>
            </div>
        );
    }
} export default PredictionList; 