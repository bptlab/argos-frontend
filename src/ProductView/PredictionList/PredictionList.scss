@import "PredictionView/PredictionView";

.prediction-list-container {
  margin-top: 20px;
  margin-bottom: 20px;
}

.prediction-list-container h2 {
  margin-bottom: 10px;
  width: 100%;
}