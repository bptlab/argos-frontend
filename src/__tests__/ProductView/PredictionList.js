import React from 'react';
import PredictionList from '../../ProductView/PredictionList/PredictionList.js';
import renderer from 'react-test-renderer';
import testConfigurations from '../testData/frontend_configurations.js';


test('Render DetailArea', () => {
    const component = renderer.create(
        <PredictionList
            configuration={testConfigurations.CONFIGURATIONS[0]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});