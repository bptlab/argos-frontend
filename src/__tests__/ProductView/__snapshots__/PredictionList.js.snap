exports[`test Render DetailArea 1`] = `
<div
  className="container prediction-list-container">
  <h2>
    Possible product errors
  </h2>
  <ul
    className="list-group">
    <a
      className="prediction-view"
      data-target="#prediction-view-EF-254-modal"
      data-toggle="modal"
      onClick={[Function]}>
      <li
        className="list-group-item justify-content-between list-group-item-action">
        EF-254
         - 
        Heating produces strange sounds
        <span
          className="badge badge-default badge-pill">
          2
        </span>
        <div
          className="prediction-view">
          <div
            className="prediction-view-EF-254 modal fade"
            id="prediction-view-EF-254-modal"
            role="dialog"
            tabIndex="-1">
            <div
              className="modal-dialog modal-lg"
              role="document">
              <div
                className="modal-content">
                <div
                  className="modal-header">
                  <h5
                    className="modal-title"
                    id="prediction-view-EF-254-label">
                    Prediction Details: Heating produces strange sounds
                  </h5>
                  <button
                    className="close"
                    data-dismiss="modal"
                    type="button">
                    <span
                      id="close">
                      Ã—
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body">
                  <h2>
                    EF-254
                  </h2>
                </div>
                <div
                  className="modal-footer">
                  <button
                    className="btn"
                    disabled="disabled"
                    onClick={[Function]}
                    type="button">
                    <div>
                      Close
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </li>
    </a>
  </ul>
</div>
`;
